### PR TITLE
JSON updates + minor rewording txzchk usage

### PR DESCRIPTION
**Pull request by xexyl**
See the git log for commits 0204642, 0f72fcc, b8807a0 and 9efaaf4a7.

I'm not sure if this is all I'll do code wise today but most likely it is. I do hope to do more reading and if I do I might in fact make some changes but these changes are at least in. I still think I need to work out a proper struct/union (possibly more than one) before I can really start writing the parser but this will come later on (hopefully sooner than later)

### CODE REVIEW

**Comment by xexyl** [comment]
I'm aware of a compilation issue on CentOS ... working on that. It will involve making `quiet` no longer a static bool but one that can be used in all the tools. I could just add `-Wno-` to the compiler for `jparse` but I think it might be better to make `quiet` like the other related variables. Even so I want to be going soon so I might resort to doing just that and keep the other idea for later.

**Comment by xexyl** [comment]
Okay now I am taking a break. Not sure what else I'll end up doing. It's possible this is it but I do hope to do a bit more later on before the person (a family member) calls here to fix some things at the house.

**Comment by xexyl** [comment]
Thank you for merging. I don't think I'll be getting anything else done today. I hope to get more done tomorrow.

I did think that waiting on seeing what you have in mind with `struct string` will also be of value for me.

Anyway have a nice day!

**Comment by lcn2** [comment]
> It will involve making quiet no longer a static bool but one that can be used in all the tools.

That seems like a good idea, @xexyl 

> Thank you for merging. I don't think I'll be getting anything else done today. I hope to get more done tomorrow.

You are most welcome, sir!

**Comment by xexyl** [comment]
> > It will involve making quiet no longer a static bool but one that can be used in all the tools.
> 
> That seems like a good idea, @xexyl

I didn't end up doing it because (as I think I noted) `iocccsize` also uses it and I don't want to modify it. I get the feeling that one is more of a read-only for all others but you - except maybe typo fixes.

If you want to do it though I think it'd be a good idea too.


> 
> > Thank you for merging. I don't think I'll be getting anything else done today. I hope to get more done tomorrow.
> 
> You are most welcome, sir!

Thanks. Have a great day!

**Comment by lcn2** [comment]
> If you want to do it though I think it'd be a good idea too.

We pushed out commit f0a6bc5ea885ffcc3e13b0824756aa4bc7aa19f9.

You were kind enough to hint at the **bogons** surrounding the dbg facility.  Hopefully with the release of dbg version 1.7 2022-04-02, most of those have been addressed.

Feel free to adjust the `-Wno-foo` flags you need, @xexyl, based on this recent change.

FYI: commit f0a6bc5ea885ffcc3e13b0824756aa4bc7aa19f9 did the following:

```c
    Release dbg version 1.7 2022-04-02

    Fixed dbg external variable bogosity.

    For far too long, the dbg facility forced the calling programs to
    declare and initialize dbg facility variables.  That might have
    been OK (but not great) when there was only the verbosity_level.

    While this is not intended an excuse, for those who wondered how
    things reached this state:

        While global variables should generally by avoided, one generally
        good exception is in support of a debugging facility such as
        dbg.  The excuse of wanting to avoid global variables might
        have been excused in early versions of dbg.  However as
        functionality of the dbg facility evolved, this excuse pushed
        too much of the complexity of silencing various aspects of the
        dbg into the calling programs where is did not belong.

    In version 1.7 2022-04-02 certain variables are now declared global
    and initialized in dbg.c.  The dbg facility now uses these globals:

        int verbosity_level = DBG_DEFAULT;     /* debug level set by -v */
        bool msg_output_allowed = true;        /* false ==> disable output from msg() */
        bool dbg_output_allowed = true;        /* false ==> disable output from dbg() */
        bool warn_output_allowed = true;       /* false ==> disable output from warn() and warnp() */
        bool err_output_allowed = true;        /* false ==> disable output from err() and errp() */
        bool usage_output_allowed = true;      /* false ==> disable output from vfprintf_usage() */
        bool msg_warn_silent = false;          /* true ==> silence msg(), warn(), warnp() if verbosity_level == 0 */

    The calling program is free to modify these dbg global variables as they please
    and when they please.  The dbg facility will use those values at the time dbg functions
    are called.

    The dbg_test code (when compiled with -DDBG_TEST) will output the state of the those globals.

    The dbg_test now simulates calls to warn(), warnp(), warn_or_err(), and warnp_or_errp()
    before calling err() or errp().

    Also in this commit are updates / improvements to man pages regarding -q and -q side effects.
```

Corrections and improvements, welcome.

**Comment by lcn2** [comment]
> I didn't end up doing it because (as I think I noted) iocccsize also uses it and I don't want to modify it. I get the feeling that one is more of a read-only for all others but you - except maybe typo fixes.

We do want to minimize the changes to `iocccsize.c`, `iocccsize.h`, and `rule_count.c` for now.  However:

```
FYI: We have, from our side, agreed to disagree with the original author of `iocccsize.c`. 
While we don't want to call attention to this fact (which is why we are not doing an "@"
name of the author), and we appreciate the original idea of `iocccsize`, our use case and
the author's intend of  his `iocccsize` differ.

As an example, the original author of `iocccsize.c` rejected creating `rule_count.c` and
wanted that function to remain inside `iocccsize.c` so that his Makefile could compile the
code from a single C file.

The original author's Makefile could not be used as a standalone Makefile as it appeared to
depend on certain variables and/or state being set before invoking his Makefile.

The original `iocccsize.c` was an Obfuscated C Code winning entry.  However this repo tried
to be as un-obfuscated as possible. The original author did not want to make any changes to
the C code that did not fit into original author's style.  While this may be a useful convention,
the fact that original author of `iocccsize.c` was being used inside this repo put the repo's
style and his style into conflict.  

For this reason we closed our pull request on his repo and added the following message:

> You are free to accept or reject the pull request, of course, as you wish. While some of
> the items you mention  are interesting, there are enough differences that won't be implemented
> in our mkiocccentry repo so as to not warrant a further merge.
```

We don't want to impugn the original author of iocccsize in any way.  We simply have widely difference use cases to not warrant an attempted code merge.   We don't want the And so original author of iocccsize to feel bad, so we plan to delete the comment after your read it.

We are tempted to comment and de-obfuscate  `iocccsize.c`, `iocccsize.h`, and `rule_count.c`,
that is **NOT a priory** at this time.  If we do some sort of code cleanup, we also want to be able to
fold in future code changes from his side should there be a need to _bug fix_ the code due to someone
exploiting a flaw in `iocccsize` (and perhaps getting an "_abuse of the rules award_" 😎) or the
ANSI C committee introducing changes to C that we are forced to react to in some way.

The `iocccsize` tool is NOT intended to be perfect in size counting.   As the `rule_count.c` comment says:

```c
/*
 * HINT: The algorithm implemented by this code may or not be obfuscated.
 *       The algorithm may not or may appear to be obfuscated.
 *
 * In particular:
 *
 *      We did not invent the algorithm.
 *      The algorithm consistently finds Obfuscation.
 *      The algorithm killed Obfuscation.
 *      The algorithm is banned in C.
 *      The algorithm is from Bell Labs in Jersey.
 *      The algorithm constantly finds Obfuscation.
 *      This is not the algorithm.
 *      This is close.
 */
```

🤓

In short, the `iocccsize` code in intended to be exactly what it is .. so perhaps instead of a "_code cleanup_",
a "_programmers apology_" with a comment near the top might be in order. 

If you can click the 👍 button or 👀 button when you read this, then we will delete it.  Please do not include this comment in any replies.

**Comment by xexyl** [comment]
> > If you want to do it though I think it'd be a good idea too.
> 
> We pushed out commit [f0a6bc5](https://github.com/ioccc-src/mkiocccentry/commit/f0a6bc5ea885ffcc3e13b0824756aa4bc7aa19f9).
> 
> You were kind enough to hint at the **bogons** surrounding the dbg facility. Hopefully with the release of dbg version 1.7 2022-04-02, most of those have been addressed.
> 

Well that does indeed sound like me! 

I will have to look at the changes more when at the laptop which won’t be until tomorrow. Unfortunately not all the work could be done today due to cooler weather (a rarity here) so more activity here tomorrow as well and there already was the plan of another family member being here. 

Hopefully this isn’t too distracting but as I said yesterday I think it was I probably wouldn’t  be able to do much more after they get here anyway. 


> Feel free to adjust the `-Wno-foo` flags you need, @xexyl, based on this recent change.
> 

Thanks. Hopefully it won’t be a problem. Nothing comes to mind atm anyway except maybe to remove the one I added today for CentOS. Will test that out tomorrow. 

> FYI: commit [f0a6bc5](https://github.com/ioccc-src/mkiocccentry/commit/f0a6bc5ea885ffcc3e13b0824756aa4bc7aa19f9) did the following:
> 
> ```c
>     Release dbg version 1.7 2022-04-02
> 
>     Fixed dbg external variable bogosity.
> 
>     For far too long, the dbg facility forced the calling programs to
>     declare and initialize dbg facility variables.  That might have
>     been OK (but not great) when there was only the verbosity_level.
> 
>     While this is not intended an excuse, for those who wondered how
>     things reached this state:
> 
>         While global variables should generally by avoided, one generally
>         good exception is in support of a debugging facility such as
>         dbg.  The excuse of wanting to avoid global variables might
>         have been excused in early versions of dbg.  However as
>         functionality of the dbg facility evolved, this excuse pushed
>         too much of the complexity of silencing various aspects of the
>         dbg into the calling programs where is did not belong.
> 
>     In version 1.7 2022-04-02 certain variables are now declared global
>     and initialized in dbg.c.  The dbg facility now uses these globals:
> 
>         int verbosity_level = DBG_DEFAULT;     /* debug level set by -v */
>         bool msg_output_allowed = true;        /* false ==> disable output from msg() */
>         bool dbg_output_allowed = true;        /* false ==> disable output from dbg() */
>         bool warn_output_allowed = true;       /* false ==> disable output from warn() and warnp() */
>         bool err_output_allowed = true;        /* false ==> disable output from err() and errp() */
>         bool usage_output_allowed = true;      /* false ==> disable output from vfprintf_usage() */
>         bool msg_warn_silent = false;          /* true ==> silence msg(), warn(), warnp() if verbosity_level == 0 */
> 
>     The calling program is free to modify these dbg global variables as they please
>     and when they please.  The dbg facility will use those values at the time dbg functions
>     are called.
> 
>     The dbg_test code (when compiled with -DDBG_TEST) will output the state of the those globals.
> 
>     The dbg_test now simulates calls to warn(), warnp(), warn_or_err(), and warnp_or_errp()
>     before calling err() or errp().
> 
>     Also in this commit are updates / improvements to man pages regarding -q and -q side effects.
> ```
> 
> Corrections and improvements, welcome.

I will look at this more carefully tomorrow but from the log it looks good and makes sense.

**Comment by xexyl** [comment]
Thank you for the interesting comments. Feel free to delete it now. 

Sorry about not reacting to as you like. I don’t see how to do it on mobile Safari but since you want to get it deleted I figured I would let you know anyway. 

All good on my end!

**Comment by xexyl** [comment]
You deleted the wrong comment so I've gone ahead and deleted the right one. I'm not sure if the other comment was important but since it was about debug facility changes I guess it's not so that's okay.

**Comment by xexyl** [comment]
Here's what I wrote about the debug facility but changed as I have verified that the CentOS warning is no longer triggered.

> You were kind enough to hint at the **bogons** surrounding the dbg facility. Hopefully with the release of dbg version 1.7 2022-04-02, most of those have been addressed.
>
Well that does indeed sound like me!

I will have to look at the changes more when at the laptop when I'm more awake.

Unfortunately not all the work could be done yesterday due to cooler weather (a rarity here) so more activity here today as well and there already was the plan of another family member being here.

Hopefully this isn’t too distracting but as I said (the other day) I probably wouldn’t be able to do much more after the time they get here even if they weren't coming here, anyway.




 
> Feel free to adjust the `-Wno-foo` flags you need, @xexyl, based on this recent change.
> 

With the commits I was able to remove the `-Wno-` warning I added yesterday. I've also fixed some typos but I've not done any commits yet.

> FYI: commit [f0a6bc5](https://github.com/ioccc-src/mkiocccentry/commit/f0a6bc5ea885ffcc3e13b0824756aa4bc7aa19f9) did the following:
> 
> ```c
>     Release dbg version 1.7 2022-04-02
> 
>     Fixed dbg external variable bogosity.
> 
>     For far too long, the dbg facility forced the calling programs to
>     declare and initialize dbg facility variables.  That might have
>     been OK (but not great) when there was only the verbosity_level.
> 
>     While this is not intended an excuse, for those who wondered how
>     things reached this state:
> 
>         While global variables should generally by avoided, one generally
>         good exception is in support of a debugging facility such as
>         dbg.  The excuse of wanting to avoid global variables might
>         have been excused in early versions of dbg.  However as
>         functionality of the dbg facility evolved, this excuse pushed
>         too much of the complexity of silencing various aspects of the
>         dbg into the calling programs where is did not belong.
> 
>     In version 1.7 2022-04-02 certain variables are now declared global
>     and initialized in dbg.c.  The dbg facility now uses these globals:
> 
>         int verbosity_level = DBG_DEFAULT;     /* debug level set by -v */
>         bool msg_output_allowed = true;        /* false ==> disable output from msg() */
>         bool dbg_output_allowed = true;        /* false ==> disable output from dbg() */
>         bool warn_output_allowed = true;       /* false ==> disable output from warn() and warnp() */
>         bool err_output_allowed = true;        /* false ==> disable output from err() and errp() */
>         bool usage_output_allowed = true;      /* false ==> disable output from vfprintf_usage() */
>         bool msg_warn_silent = false;          /* true ==> silence msg(), warn(), warnp() if verbosity_level == 0 */
> 
>     The calling program is free to modify these dbg global variables as they please
>     and when they please.  The dbg facility will use those values at the time dbg functions
>     are called.
> 
>     The dbg_test code (when compiled with -DDBG_TEST) will output the state of the those globals.
> 
>     The dbg_test now simulates calls to warn(), warnp(), warn_or_err(), and warnp_or_errp()
>     before calling err() or errp().
> 
>     Also in this commit are updates / improvements to man pages regarding -q and -q side effects.
> ```
> 
> Corrections and improvements, welcome.

I will have to look at these later but I am sure it's fine. I won't know until I either add more code involving debug output or else I add new code involving debug output. Both are likely to happen but I don't know which one will happen first.

Based on how I feel right now I would say I won't get much done today. However it's still early and who knows what might happen in a bit if I can wake up better.

Hope you're having a good sleep!

**Comment by lcn2** [comment]
> You deleted the wrong comment so I've gone ahead and deleted the right one. I'm not sure if the other comment was important but since it was about debug facility changes I guess it's not so that's okay.

Sorry (tm Canada 🇨🇦)

**Comment by xexyl** [comment]
> > You deleted the wrong comment so I've gone ahead and deleted the right one. I'm not sure if the other comment was important but since it was about debug facility changes I guess it's not so that's okay.
> 
> Sorry (tm Canada 🇨🇦)

Happens to everyone!
